### PR TITLE
📖 Simplify getting-started.md and remove broken links

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**: The cluster API quickstart points to this getting-started doc for setting up the Azure service principal. This removes all extra information that might confuse first time users and refreshes the content to be clear and concise.
 
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
Simplify getting-started.md and remove broken links
```

### CODE REVIEW

File: docs/getting-started.md [hunk 1]
```diff
@@ -1,72 +1,48 @@
...
+
+  4. Save your Subscription ID in an environment variable.
 
   ```bash
   export AZURE_SUBSCRIPTION_ID="<SubscriptionId>"
   ```
 
-  3. Create an Azure Service Principal by running the following command or skip this step and use a previously created Azure Service Principal.
+  5. Create an Azure Service Principal by running the following command or skip this step and use a previously created Azure Service Principal.
+  NOTE: the "owner" role is required to be able to create role assignments for [system assigned managed identity](book/src/topics/identity.md).
```
Comment (mboersma) [inline docs/getting-started.md:39]:
```suggestion
  NOTE: the "owner" role is required to be able to create role assignments for [system-assigned managed identity](book/src/topics/identity.md).
```

**Review by mboersma — CHANGES_REQUESTED** [review]
Looks great, just had one little grammar consistency nit to pick.

**Review by mboersma — APPROVED** [review]
lgtm

**Review by devigned — APPROVED** [review]
/lgtm
/approve

**Pull request merged**